### ISSUE TITLE
Change `process_runtime_heap_alloc_bytes` metric to include memory ballast size

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector#8342]
The [`process/runtime/heap_alloc_bytes`][1]/[`process_runtime_heap_alloc_bytes`][2] metric is defined as 
> Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')

but, since #45, the actual value is `runtime.MemStats.HeapAlloc` minus the memory ballast size if the `memory_ballast` extension is being used:

https://github.com/open-telemetry/opentelemetry-collector/blob/7030388dde3f765af60cf8431ea54340194d9f18/service/internal/proctelemetry/process_telemetry.go#L283-L293

This delays the process metrics initialization until extensions are running: https://github.com/open-telemetry/opentelemetry-collector/blob/7030388dde3f765af60cf8431ea54340194d9f18/service/service.go#L223

We should remove this special case and change the metric to report the raw value of `runtime.MemStats.HeapAlloc` instead, which will simplify both the metric's semantics as well as the implementation of telemetry initialization. If we keep using the memory ballast extension (see the separate issue #8343 for discussion on this, which may happen after this change), we can add a metric to the `memory_ballast` extension to indicate the memory ballast size and be able to recover the previous value reported by this metric.

If we find out it is not desirable to just report `runtime.MemStats.HeapAlloc`, we should instead change the metric description to state that the memory ballast value is removed.

[1]: https://github.com/open-telemetry/opentelemetry-collector/blob/7030388dde3f765af60cf8431ea54340194d9f18/service/internal/proctelemetry/process_telemetry.go#L119-L122
[2]: https://github.com/open-telemetry/opentelemetry-collector/blob/7030388dde3f765af60cf8431ea54340194d9f18/service/internal/proctelemetry/process_telemetry.go#L186-L193

**Comment by mx-psi** [comment]
> This change would make sense to me. How do you propose introducing the change? I'd expect anyone that monitors using this value today will be surprised to see that value change by the ballast size after a release.

We can add the `memory_ballast` metric without breaking anyone. For the `process_runtime_heap_alloc_bytes`, we can do the change through a feature gate to allow users to migrate at their own pace.

**Comment by codeboten** [comment]
Sounds good. Pinging @open-telemetry/collector-approvers for more eyes on this

**Comment by dmitryax** [comment]
Why do we need to tackle this before the potential removal of the memory_ballast? I think doing it implicitly for user is pretty tricky.

I'd suggest not touching it until we decide on the removal of the memory_ballast. I hope we will be able to remove the memory ballast. In that case, the transition will be pretty obvious for the user:
1. memory_ballast is deprecated: users explicitly remove it and get `process_runtime_heap_alloc_bytes` updated
2. memory_ballast is removed: users have to remove it from their config by this time and get `process_runtime_heap_alloc_bytes` updated

**Comment by codeboten** [comment]
@dmitryax right I guess this is currently blocked on the decision out of https://github.com/open-telemetry/opentelemetry-collector/issues/8343

**Comment by mx-psi** [comment]
> Why do we need to tackle this before the potential removal of the memory_ballast? I think doing it implicitly for user is pretty tricky.

Let's discuss #8343 first, but my expectation would be that removing the memory ballast is going to be a lengthy process given how widely it is used, and I don't want to block doing this refactor on the ballast removal because
1. this issue is relevant for the latest approach I am trying with #4970 
2. I would want this issue to happen even if we don't remove the memory ballast or don't do #4970; the current metric description is wrong and we privilege this ballast extension implementation above other future approaches for no good reason other than the historical one.

**Comment by mx-psi** [comment]
We can wait for open-telemetry/opentelemetry-helm-charts/issues/891, if we can remove the extension 'quickly' I think we can avoid doing this in two steps.
